### DISCUSSION TITLE
how to chain SagerNet + OpenVPN

### DISCUSSION THREAD

**Discussion by klcheungaj** [SagerNet/SagerNet#668] [Q&A]
I would like to get out of gfw using sagernet, and then run the openvpn client to connect to the server. 
What is the correct way to do it?

**Comment by siq0o** [comment]
You can only do this using a tcp connection. turn sagernet into proxy mode only. add proxy in your openvpn config (or from the menu).

**Reply by klcheungaj** [reply]
hi, 
I did it before but still failed. Can you please help to check if my configuration is correct? 
I set sagernet to proxy mode and enabled HTTP inbound. In openvpn client proxy setting, I used IP 127.0.0.1 and port 9080. No username and password. I also tried to add "http-proxy 127.0.0.1 9080" directive to the openvpn profile.
